### ISSUE TITLE
Foreign Keys not named correctly on code generation

### ISSUE THREAD

**Issue by Falthazar** [msawczyn/EFDesigner#15]
Similar to my last one, sort of.

Same three tables:
![image](https://user-images.githubusercontent.com/19520197/44301640-81e23c80-a2e8-11e8-8084-dc154a5baae8.png)

in the context, it generates:
```Csharp
modelBuilder.Entity<EFCoreDesignTools.JoinSiteInvestigator>()
    .HasOne(x => x.Site)
    .WithOne()
    .HasForeignKey("Site_Id")
    .IsRequired().OnDelete(DeleteBehavior.Cascade);
modelBuilder.Entity<EFCoreDesignTools.JoinSiteInvestigator>()
    .HasOne(x => x.Investigator)
    .WithOne()
    .HasForeignKey("Investigator_Id")
    .IsRequired().OnDelete(DeleteBehavior.Cascade);
```
`.HasForeignKey("Site_Id")` and `.HasForeignKey("Investigator_Id")` should be generated as "SiteId" and "InvestigatorId" respectively.

I'm not sure if this is an issue with how the relationships are set up, or an issue with the generator?

**Comment by msawczyn** [comment]
It's how you have them set up. :-)

The designer will create foreign keys on its own, using Entity Framework's naming conventions. That convention is <property name>_Id, so it's creating what the EF team calls _shadow properties_ to handle that. 

Many to many associations in EFCore are problematic (read: not yet supported) and, since the Entity Framework team will eventually handle those, we haven't made an effort to try to detect a manual many-to-many design knowing that it would (should?) be quite temporary and that the workaround code would later be removed.

**Comment by msawczyn** [comment]
Yes, EF6 will create the needed database structures without messing up your object model. EFCore will _eventually_ get there, but it's slow going. 

The ideal case is that you model your classes like you would model any classes. Persistence is something EF takes care of, and you shouldn't really have to care how it does it. That's why I rail against putting foreign key properties in a class -- you wouldn't put them there if it weren't persistent, would you? -- and in using the LINQ `Join` statement ... too databasey.

EFDesigner's philosophy is that you're modeling classes, not tables. Sure, there are a few properties that point to the fact that persistence is done via database tables, but that's because we don't live in a perfect world. Except for the connection string and _maybe_ one or two other properties, the defaults serve well enough that you shouldn't really have to mess with them.

I use [Tangible's editor](https://marketplace.visualstudio.com/items?itemName=tangibleengineeringGmbH.tangibleT4Editor240plusmodelingtoolsforVS2017). It says that it doesn't like string interpolation (the $"string" thing), but that doesn't matter ... at runtime it just works, since the .NET version at runtime is more up-to-date than then one they use at edit time. (They should update that.)
